lib/features/booking/widgets/update_phone_no_bottom_sheet_view.dart

```diff
@@ -112,11 +180,82 @@ class _UpdatePhoneNoBottomSheetBody
                             top: HsDimens.spacing8,
                             left: HsDimens.spacing16,
                             right: HsDimens.spacing16),
-                        child: PrimaryButton(
-                          label: HatSpaceStrings.current.save,
-                          // TODO: validate On Press event
+                        child: ValueListenableBuilder<PhoneNumberErrorType?>(
+                          valueListenable: _phoneNumberError,
+                          builder: (context, value, child) {
+                            return PrimaryButton(
+                                label: HatSpaceStrings.current.save,
+                                onPressed: _phoneNumber == null
+                                    ? null
+                                    : value?.phoneNumberError != null
+                                        ? null
+                                        : () {
+                                            // TODO: SAVE AND UPLOAD TO DATABASE
+                                          });
+                          },
                         )))
               ],
             )));
   }
 }
+
+class PhoneNumberInputFormatter extends TextInputFormatter {
+  final ValueNotifier<PhoneNumberErrorType?> error;
+
+  PhoneNumberInputFormatter(this.error);
+  @override
+  TextEditingValue formatEditUpdate(
+      TextEditingValue oldValue, TextEditingValue newValue) {
+    // FORMAT PHONE NUMBER
+    if (oldValue.text.length < newValue.text.length) {
+      if (newValue.text.length == 5) {
+        return TextEditingValue(
+            text:
+                '${oldValue.text.substring(0, 4)} ${newValue.text.substring(newValue.text.length - 1)}');
+      }
+
+      if (newValue.text.length == 9) {
+        return TextEditingValue(
+            text:
+                '${oldValue.text.substring(0, 8)} ${newValue.text.substring(newValue.text.length - 1)}');
+      }
+    }
+
+    // Length VALIDATION
+    if (newValue.text.isEmpty ||
+        newValue.text.trim().length <
+            PhoneNumberErrorType.minLength.maxPhoneNumber) {
+      error.value = PhoneNumberErrorType.minLength;
+      return newValue;
+    }
+    if (newValue.text.length > PhoneNumberErrorType.minLength.maxPhoneNumber) {
+      return oldValue;
+    }
+
+    // AREA CODE VALIDATION
+    RegExp codeFormat = RegExp('02|03|04|05|07|08');
+    if (!newValue.text.startsWith(codeFormat)) {
+      error.value = PhoneNumberErrorType.wrongCode;
+      return newValue;
+    }
+    error.value = null;
+    return newValue;
+  }
+}
+
+enum PhoneNumberErrorType {
+  minLength(),
```

Comment:
Enum does not need () if there's no param assigned to it.

Reply:
fiX